### PR TITLE
Migrate to Readthedocs and Sphinx

### PR DESCRIPTION
**Pull request by Opt-Mucca**
This change moves our documentation from GitHub to Readthedocs.
I've added a basic structure / install guides / introduction to most plug-ins / motivation for using SCIP / FAQ / API reference manual.
I've also added a few new wrapper functions. These were needed when implementing the new examples in the documentation that are now duplicated as tests.

After this gets merged we'd need to do the following:
- Make a readthedocs account for pyscipopt
- Link it to the GitHub
- Update the README and INSTALL

### CODE REVIEW

File: docs/similarsoftware.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+- `CPLEX <https://www.ibm.com/products/ilog-cplex-optimization-studio>`_: LP, MILP, QP, MIQP, QCP, MIQCP, SOCP
+- `Gurobi <https://www.gurobi.com/>`_: LP, MILP, QP, MIQP, QCP, MIQCP, MINLP, PB, SOCP
+- `Xpress <https://www.fico.com/en/products/fico-xpress-optimization>`_: LP, MILP, QP, MIQP, QCP, MIQCP, MINLP, SOCP
+- `COPT <https://www.copt.de/>`_: LP, MILP, QP, MIQP, QCP, MIQCP, SOCP, SDP
+- `MOSEK <https://www.mosek.com/>`_: LP, MILP, QP, MIQP, QCP, MIQCP, SOCP, SDP, MISDP
+
+General Modelling Frameworks (Solver Agnostic)
+==============================================
+
+This list will contain general modelling frameworks that allow you to use swap out SCIP for other
```
Comment (Joao-Dionisio) [inline docs/similarsoftware.rst:67]: "allow you to use swap out SCIP" - I don't know if this is gramatically correct.

Also, maybe add AMPL here as well?

**Comment by Joao-Dionisio** [comment]
You already know I find this absolutely incredible, Mark, and we discussed reviewing these things after merging. So on my end, we're good to go!

File: docs/similarsoftware.rst (hunk 1 above)
Comment (Opt-Mucca) [reply docs/similarsoftware.rst:67]: I wanted to keep this Python specific. So I skipped a lot of solvers and frameworks

**Comment by Opt-Mucca** [comment]
@Joao-Dionisio thank you! I am extremely happy with the outcome of this. (you have to actually approve the MR which I always forget to do)

File: docs/similarsoftware.rst (hunk 1 above)
Comment (Opt-Mucca) [reply docs/similarsoftware.rst:67]: TIL Ampl has a Python interface. I will add it

**Review by Joao-Dionisio — APPROVED** [review]

**Pull request merged**